File: 2001_3000/2108/2108.py
import sys

ndict = {i:0 for i in range(-4000,4001)}
# n 의 갯수
numOfN = int(sys.stdin.readline().rstrip())

# 사전 자료구조를 만들어서 개수를 구함
for i in range(numOfN):
    inp = int(sys.stdin.readline().rstrip())
    ndict[inp] += 1

nlist = []
# 카운팅 정렬으로 값을 넣어줌
for [key, value] in ndict.items():
    for _ in range(value):
        nlist.append(key)

# 최빈 값을 구하기 위한 리스트
orderByCommon = sorted(ndict.items(), key=lambda item:item[1], reverse=True)

# 산술 평균
print(round(sum(nlist)/numOfN))
# 중앙 값
print(nlist[numOfN//2])
# 최빈 값
if len(orderByCommon) > 1:
    if orderByCommon[0][1] == orderByCommon[1][1]:
        print(orderByCommon[1][0])
    else: print(orderByCommon[0][0])
else:
    print(orderByCommon[0][0])
# 범위
print(nlist[-1]-nlist[0])

File: 2001_3000/2108/2108.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

int main(){
    
    vector<pair<int, int>> n_dict(8002);
    vector<int> n_list;
    int N;
    long long sum = 0;

    cin>>N;

    for(int i=-4000; i<4001; i++){
        n_dict[i+4000].first = i;
        n_dict[i+4000].second = 0; 
    }

    for(int i=0; i<N; i++){
        int n;
        cin>>n;
        n_dict[n+4000].second += 1;
        sum += n;
    }

    for(pair<int, int> item : n_dict) {
        for(int i=0; i<item.second; i++){
            n_list.push_back(item.first);
        }
    }

    sort(n_dict.begin(), n_dict.end(), [](pair<int, int> a,pair<int, int> b) -> bool
    {
        if(a.second == b.second){
            return a.first > b.first;
        }
        // 내림차순
        return a.second > b.second;
    });

    cout<<sum/N<<endl;
    cout<<n_list[n_list.size()/2]<<endl;
    if(n_dict[0].second == n_dict[1].second){
        cout<<n_dict[1].first;
    } else{
        cout<<n_dict[0].first;
    }
    cout<<n_list[n_list.size()-1]-n_list[0]<<endl;
    
    return 0;
}